backend/src/platformer_handler.erl

```diff
@@ -6,17 +6,48 @@
 -export([websocket_handle/2]).
 -export([websocket_info/2]).
 
+
+
 init(Req, State) ->
-	{cowboy_websocket, Req, State}.
+	{cowboy_websocket, Req, State, #{idle_timeout => 300000}}.
 
 websocket_init(State) ->
+    broadcaster ! {subscribe, self()},
 	{ok, State}.
 
-websocket_handle({text, Json}, State) ->
-    case (jsx:is_json(Json)) of
-        true ->  Map = jsx:decode(Json),
-                 io:format("received JSON: ~w~n", [Map]),
-	             {reply, {text, Json}, State};
+% For handling casts based on decoded JSON data. 
+% Sends a response only in the case of an invalid cast.
+% TODO: Replace with console logging and never send a response
+json_cast(Json, State) ->
+    case (maps:get(<<"type">>, Json, notype)) of
+        notype -> Res = jsx:encode([{<<"error">>, <<"Must specify cast type">>}]),
+                  {reply, {text, Res}, State};
+        _      -> Res = jsx:encode([{<<"error">>, <<"unrecognized cast type">>}]),
+                  {reply, {text, Res}, State}
+    end.
+
+% For handling calls based on decoded JSON data
+json_call(Json, State) ->
+    case (maps:get(<<"type">>, Json, notype)) of
+        notype -> Res = jsx:encode([{<<"error">>, <<"Must specify call type">>}]),
+                  {reply, {text, Res}, State};
+        _      -> Res = jsx:encode([{<<"error">>, <<"unrecognized call type">>}]),
+                  {reply, {text, Res}, State}
+    end.
+
+
+% State is [] on app startup
+websocket_handle({text, Data}, State) ->
```

Comment:
Could we get some clarifying docs on exactly the form of data to send to these and how to do so? Otherwise seems great